setuptools/command/bdist_egg.py

```diff
@@ -238,13 +239,26 @@ def run(self):
 
     def zap_pyfiles(self):
         log.info("Removing .py files from temporary directory")
+        py3 = sys.version_info >= (3, 0)
+        re_pycache_file = re.compile('(.+)\.cpython-3\d(.pyc)$')
```

Comment:
Probably best not to hard-code some of the "magic tag" (i.e. `cpython-3\d`). Also, there's a bug in the use of unescaped `.` in `.pyc`. Also, I think there's an issue with this string not being a raw string (Python 3.6 has warnings about these). Maybe instead:

```
pattern = r'(?P<name>.+)\.(?P<magic>[^.]+)\.pyc'
```

Then use `m.groupdict()` or `m.group('name')` to extract the name.